components/buy/Buy.vue

```diff
@@ -236,6 +239,16 @@ const watchNftsPriceChanges = (nftIds: string[], force: boolean = false) => {
   })
 }
 
+watch(
+  () => preferencesStore.completePurchaseModal.isOpen,
+  (isOpen, prevIsOpen) => {
+    const modalGotClosed = !isOpen && prevIsOpen
+    if (modalGotClosed) {
+      nftSubscription.unsubscribe()
```

Comment:
let use this 

```suggestion
       clearNftSubscription()
```

and needs `nftSubscription.active` to false  , otherwise when you close the modal and open it again since it's already true it wont subscribe again  

````diff
const clearNftSubscription = () => {
      nftSubscription.unsubscribe()
-      nftSubscription.nftIds = [] 
+     nftSubscription.active = false
}

````